### PR TITLE
Cursor in state pt 2

### PR DESCRIPTION
**Pull request by bcolloran**
follow up from #1596 -- this is builds on the changes in that PR; @hamilton if that looked ok, take a look at this next.

this will get us almost all the way there for #1594 

### Pull Request checklist

- [ ] **Documentation**: If this feature has or requires documentation, the relevant docs have been updated.
- [ ] **Changelog**: This PR updates the [changelog](../CHANGELOG.md) with any user-visible changes.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not

no user-facing changes, no doc or changelog entry needed

### CODE REVIEW

File: src/components/codemirror-utils.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export function getAllSelections(doc) {
```
Comment (wlach) [inline src/components/codemirror-utils.js]: This feels like a function that should have at least a few tests.

Comment (bcolloran) [reply src/components/codemirror-utils.js]: i agree in principle, but i'm not sure it's best to invest in cleaning up pre-existing technical debt around code mirror in this PR (this is just one example of the zillion function in our codebase that has no tests -- in general i've tried to leave things better than i found them by adding tests around some action creators that didn't have any previously, and that's what i've been shooting for in my last few PRs -- adding a few new tests for old code, and testing  my new code)

File: src/actions/actions.js [hunk 2]
```diff
@@ -188,19 +187,44 @@ export function addLanguage(languageDefinition) {
...
     if (kernelState !== "KERNEL_BUSY") dispatch(setKernelState("KERNEL_BUSY"));
-    const cm = window.ACTIVE_CODEMIRROR;
-    const doc = cm.getDoc();
-    if (!doc.somethingSelected()) {
-      const { line } = doc.getCursor();
-      const activeChunk = getChunkContainingLine(jsmdChunks, line);
+    if (getState().editorSelections.length === 0) {
+      const activeChunk = getChunkContainingLine(
+        jsmdChunks,
+        getState().editorCursorLine
```
[the remark this answers is not in the document]
Comment (bcolloran) [reply src/actions/actions.js]: fixed

File: src/actions/actions.js [hunk 3]
```diff
@@ -188,19 +187,44 @@ export function addLanguage(languageDefinition) {
...
+      );
       evalQueue.evaluate(activeChunk, dispatch);
     } else {
-      const selectionChunkSet = getAllSelections(doc)
-        .map(selection => selectionToChunks(selection, jsmdChunks, doc))
+      const selectionChunkSet = getState()
```
[the remark this answers is not in the document]
Comment (bcolloran) [reply src/actions/actions.js]: fixed

**Comment by bcolloran** [comment]
closing this one, stupid rebasing bs....

File: src/components/codemirror-utils.js (hunk 1 above)
Comment (wlach) [reply src/components/codemirror-utils.js]: Well, I can't really force you to do anything, but I would encourage you to write at least one or two tests for this function in particular. Refactoring tends to be exactly when you break things unexpectedly, and writing tests is some insurance against that. Especially in a case like this where I suspect writing a test should only take 10 minutes or so.

**Pull request closed without merging**